[Shipping Label] Add condition checks before showing "Email" and "Message" actions

reginabally opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #7983
<!-- Id number of the GitHub issue this PR addresses. -->

## Description
If the customer address can't be verified during shipping label creation, the app will propose to contact the customer via email / message / call. Tapping on the "Email" or "Message" action does nothing if the device doesn't have Mail or Message apps configured.

This PR adds additional checks before displaying the "Email" and "Message" actions after user taps the "Contact Customer" button on the top banner.

## Testing instructions

Pre-requisite:
1. The test store must have WooCommerce Shipping & Tax plugin installed and activated.
2. Uninstall the Mail app on the device
3. Turn on the Airplane mode on the device, enable WiFi, and disable iMessage. This *should* prevent the app from being able to send out SMS.

Testing steps:
1. Create an order and collect the payment
4. Tap `Create Shipping Label` and fill in the `Ship from` data (first screen)
5. Fill the data for the `Ship to` screen with a semantically valid but non-existing address and tap `Done`
6. Once the `We were unable to verify shipping address...` alert is shown, tap the `Contact Customer` button
7. Notice the "Email" and "Messages" options will not be shown.

## Screenshots

<img src="https://user-images.githubusercontent.com/40906847/221174106-f9e29512-7f7d-440b-b7e8-0507520956ff.png" width=300/></td>

https://user-images.githubusercontent.com/40906847/221175007-df521b19-48ce-492b-884b-7b70f8e424ed.mp4


---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

reginabally committed:
Add condition checks before showing "Email" and "Message" actions.

reginabally commented:
Hi @rachelmcr, Thanks in advance for reviewing this PR! I tested this change on Xcode Simulator, where Mail and Messages apps are not available. So I haven't been able to test whether the "Email" or "Message" option will be available when both apps work.

Also, I need some guidance to address another issue reported in #7983:

> there's no animation when Contact Customer is tapped, same for Open Map

I can't seem to find a button in the app where we show animation when tapped. Can you please show me some sample screens and buttons where we show feedback when tapped? I'd like to standardize the approach for a better UX. Thanks!

rachelmcr commented on WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressFormViewController.swift:
```diff
@@ -42,7 +42,7 @@ final class ShippingLabelAddressFormViewController: UIViewController {
 
             actionSheet.addCancelActionWithTitle(Localization.contactActionCancel)
-            if let email = email, email.isNotEmpty {
+            if let email = email, email.isNotEmpty && MFMailComposeViewController.canSendMail() {
```
nit: You can use `if let email` now (as of Xcode 14), instead of `if let email = email`.

rachelmcr reviewed (commented):
This works as described! I tested on a simulator and on a real device. On my device (with phone, messaging, and email apps set up) I had all of the expected options, where as on the simulator I only had the "Call" option.

That leads me to another question, though: Should we also add a check for the "Call" option, so it only appears if the device can make a call?

rachelmcr commented:
> Also, I need some guidance to address another issue reported in https://github.com/woocommerce/woocommerce-ios/issues/7983:

>> there's no animation when Contact Customer is tapped, same for Open Map

🤔 I'm not entirely sure what animation they were expecting there. It might be worth asking for clarification on the issue first, to make sure we know what was expected.

reginabally commented:
Thanks for reviewing this PR, @rachelmcr!

> That leads me to another question, though: Should we also add a check for the "Call" option, so it only appears if the device can make a call?

I've considered that, but if a device couldn't send emails and messages and was not even able to initiate a call, we would have nothing to show when the user taps the "Contact Customer" button.

The possible approaches I can think of when none of the three actions (Email, Message, Call) are available on the device are:

* Displays an error when "Contact Customer" is tapped
* Remove the "Contact Customer" button from the top banner

What do you think? 

Edited to add a screenshot for the first approach:

<img src="https://user-images.githubusercontent.com/40906847/222098413-07a169fc-8159-41c3-8a6e-9f29a90222ca.png" width=300 />

rachelmcr commented:
Good thought! I think either of those approaches would work. I'd lean slightly toward removing the button (since we know it won't do anything) but I don't feel super strongly about that. Since this is probably a rare scenario, it isn't worth agonizing over. I think go with whichever approach feels appropriate to you and straightforward to implement.

reginabally committed:
Add additional checks for supported contact methods.

reginabally committed:
Squashed commit of the following:

commit 95cea85df64214af4a112e9a461e49768d0188f1
Merge: 94e12b26a9 b8476192a4
Author: Rachel McR <rachel@automattic.com>
Date:   Wed Mar 1 08:58:23 2023 +0000

    Merge pull request #8985 from woocommerce/issue/8936-downloadable-files-actions

    SPE: Always include Downloadable Files action for simple products

commit 94e12b26a9e0068e961e4be2ae0849cb9f91a7fe
Merge: 225b907c1c e416f69178
Author: Rachel McR <rachel@automattic.com>
Date:   Wed Mar 1 08:56:17 2023 +0000

    Merge pull request #8984 from woocommerce/issue/8936-downloadable-files-settings

    SPE: Remove downloadable setting from product settings

commit 225b907c1cc8b4f98779e39c1f92ca98c8922b55
Merge: 0068549ed1 0e7af503dc
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Wed Mar 1 12:34:59 2023 +0530

    Merge pull request #9005 from woocommerce/feat/8970-onboarding-survey

    [Store Onboarding] Feedback for store setup

commit 0068549ed19258013586e0e8dd71ef646b7cc8d7
Merge: d0af7e586d d3556f0cce
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Wed Mar 1 11:15:56 2023 +0530

    Merge pull request #8991 from woocommerce/feat/8907-polish-store-setup-list

    [Store Onboarding] Polish store setup card UI

commit 0e7af503dcc148000e94374d6c56e2acd416fe10
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Wed Mar 1 11:05:30 2023 +0530

    Pass site to `StoreOnboardingCoordinator`

commit 387f81a5dbe954af90d05f26cb7fec38d1b86885
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Wed Mar 1 08:22:00 2023 +0530

    Remove extra comment line.

commit 6a7f5c0fefae8bdcbed475e1cce8a31a613635fd
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 17:43:52 2023 +0530

    Present the store setup survey.

commit a31d0922057129275511f277858afe80d2a2b638
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 17:40:36 2023 +0530

    Add action closure for share feedback.

commit 8b120b1dcdcc3725183fa6ae760612631eeb9460
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 12:28:56 2023 +0530

    Show action sheet with "Share feedback" option.

commit 22be9c7324dd91dc396516bc2a38d1b643fbe76d
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 12:24:22 2023 +0530

    Feedback context for store setup survey.

commit df01384cac2368828b27d1ab3ff022c4d46c4a23
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 12:23:50 2023 +0530

    Test correct survey link is loaded.

commit ce7260745ea554f0098d2163b51dd9562abfcddf
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 12:18:57 2023 +0530

    Add store setup as a survey case.

commit 6db1e59476cfe651962e5bc486a2ca3308230e53
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 12:16:40 2023 +0530

    Add URL for store setup feedback.

commit d3556f0cce4a6dd857104a3d91798a72573cbf35
Merge: 73413fbbbd d0af7e586d
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Wed Mar 1 10:43:57 2023 +0530

    Merge branch 'trunk' into feat/8907-polish-store-setup-list

commit d0af7e586d6e9bf9caece25f4a0869c6f969c024
Merge: 6aa295a0a8 2e3ee75cc2
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Wed Mar 1 13:01:53 2023 +0800

    Merge pull request #8967 from woocommerce/feat/8909-onboarding-custom-domains

    Store onboarding > custom domains task: navigate to custom domains

commit 6aa295a0a86264ba7bbd603c8f0923529e8f58d9
Merge: 6e839b40a5 bc621ee76c
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Wed Mar 1 11:59:02 2023 +0700

    Merge pull request #9001 from woocommerce/feat/8918-check-passwordless-account

    REST API: Check passwordless account for the input WPCom email

commit bc621ee76c039959c2738b3fc04752d5c7ee5ea2
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Wed Mar 1 11:17:50 2023 +0700

    Point WordPressAuthenticator to trunk

commit 73413fbbbde53c24d7cb9ae1cce0030c6a344f48
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Wed Mar 1 09:41:11 2023 +0530

    Apply background color only for collapsed mode.

commit 2e3ee75cc2e5b7d1110b12c09be71b0f8106b1b2
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Wed Mar 1 09:11:59 2023 +0800

    Add a cancel CTA to domain settings modal since swipe-to-dismiss gesture does not work in landscape mode.

commit b75cf743f350b149e2745a7a34d82daeb129db5d
Merge: 1732c3bdba 6e839b40a5
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Wed Mar 1 09:11:04 2023 +0800

    Merge branch 'trunk' into feat/8909-onboarding-custom-domains

    * trunk: (115 commits)
      8992 Fix for IPP banner display edge case
      8992 Test historic IPP poweruser banner case
      8992 Remove unneccesary constant
      8992 Show OrderCreation banner when no IPP survey
      Rename reloadRoles to checkIfUserGotSufficientRole
      Keep weak self in retrieve user callback closure
      Use example.com instead of test.com for dummy domains
      8992 add IPP feedback banner logic fix to 12.6 release
      8992 Prevent firstIPP banner from showing for old orders
      8992 Improve display of feedback banners
      8992 Push the feedback banner to the viewcontroller
      Removed outdated warning about VM recreation
      8992 remove unnecessary survey completionHandler
      8992 Require receipt_url for all orders wrt IPP surveys
      8992 Update tests to include datePaid for recency
      8992 Fix arrangement of receipt_url in tests
      8992 Remove duplictated `hasTransactions` conditoin
      8992 Only require COD for the first survey
      Always show WPCom email login after getting sufficient permission
      8992 Arrange tests of firstTransaction/powerUser surveys
      ...

commit 6e839b40a5a136409698c0f26767b780fc410592
Merge: c0bbfcfc50 9f8721f688
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Tue Feb 28 14:55:42 2023 +0000

    Merge pull request #8994 from woocommerce/issue/8992-fix-ipp-survey-logic

    [Mobile Payments] Fix IPP feedback survey banner display logic

commit 9f8721f6884887a3b84224a2e02eac313963c625
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Tue Feb 28 13:57:52 2023 +0000

    8992 Fix for IPP banner display edge case

    When an IPP banner had been previously displayed, then a change in the underlying data led to an error, orderCreation, or no banner showing, the banner would not show again even when the underlying data got back to normal.

    This change preserves the block on tracking this as a new display (as it can happen several times in quick succession) while ensuring the banner is shown again.

commit 077ea53c5a6e9f6ecac323731301166304777f4c
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Tue Feb 28 13:42:56 2023 +0000

    8992 Test historic IPP poweruser banner case

commit 2254b1ca4f0576b895a4f9ee6e3bbbe74d21937b
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Tue Feb 28 13:37:13 2023 +0000

    8992 Remove unneccesary constant

commit cefb0ba71aa11ce1c8646739120bf8600685f171
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Tue Feb 28 13:32:39 2023 +0000

    8992 Show OrderCreation banner when no IPP survey

    By allowing `none` for the IPP banner source, we accidentally prevented the OrderCreation banner from showing, as we could “show” an empty IPP banner. This didn’t appear on screen, but dropped out of the top banner presentation code before the order creation banner got a chance to show.

    This change removes the optionality from the TopBanner enum, so we either have an IPP banner to show, or we don’t (and we move to Order Creation), no half-way house.

    This also fixes an issue where an (incorrectly shown) COD banner was not always cleared when all the data was fetched.

commit 169a4d9021b8c10a26847a8c0d5f7363e0898879
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Tue Feb 28 19:14:45 2023 +0700

    Add more tests for JetpackSetupCoordinator

commit 328511d9e110325ca3cbe6f8d6cf7ccb4f36658e
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 17:14:21 2023 +0530

    Use a scrollview when the onboarding view is shown in expanded state.

commit 9751b0c1b14a2c61c13cabadeba020c5db5b9c82
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Tue Feb 28 18:37:37 2023 +0700

    Update comment for JetpackSetupCoordinator

commit f3fef5e655139d8a02972da2ef5f2f4b50d31e13
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 16:50:41 2023 +0530

    Introduce `StoreOnboardingViewHostingController` and add a navigation bar button for dismissing view.

commit 04d953f85675d8c01a21cff5e3202a54ca6603be
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Tue Feb 28 18:17:39 2023 +0700

    Make the callback closure of WPComEmailLogin async to show loading on the primary CTA

commit 4958e8e846765dac2a503aab356c70ca05eb38d8
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Tue Feb 28 17:33:56 2023 +0700

    Integrate WPCom account check

commit 3b1f8c7fb52f9e836395743fe39e5b60971b86a5
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Tue Feb 28 16:16:31 2023 +0700

    Update authenticator version

commit c0bbfcfc50b494b0a672f77cb7fc00a2c283d29b
Merge: 4147394de4 fd0c7fff1f
Author: Rachel McR <rachel@automattic.com>
Date:   Tue Feb 28 09:11:58 2023 +0000

    Merge pull request #8980 from woocommerce/issue/8745-product-details-disclosure-indicators

    [Products] Show disclosure indicators in product detail rows when row is actionable

commit ddf16042bb10ae032227c456175b2e9ef0bd8740
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Tue Feb 28 16:06:35 2023 +0700

    Point WordPressAuthenticator to a new version

commit 09f52cd10829d24a50026e77110c0dff0fe9faaf
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Tue Feb 28 16:06:20 2023 +0700

    Move WordPressAuthenticator initialization to an extension for reusability

commit fd0c7fff1f74fb5a93e5e0800ef4125c02e40485
Merge: 25e9daab01 4147394de4
Author: Rachel McR <rachel@automattic.com>
Date:   Tue Feb 28 08:35:04 2023 +0000

    Merge branch 'trunk' into issue/8745-product-details-disclosure-indicators

commit 4147394de46d81da091a26b04c500ddbd4916d84
Merge: 9973810546 a4b2c89542
Author: Rachel McR <rachel@automattic.com>
Date:   Tue Feb 28 08:33:37 2023 +0000

    Merge pull request #8978 from woocommerce/issue/8947-product-details-categories-subtitle

    SPE: Update placeholder when no product category selected

commit 99738105463542dca56ecbc5b759516c0fe58e3b
Merge: 6b580f7ea8 8327b035bc
Author: Rachel McR <rachel@automattic.com>
Date:   Tue Feb 28 08:33:20 2023 +0000

    Merge pull request #8966 from woocommerce/issue/8837-spe-close-modal-after-save

    SPE: Dismiss modal and navigate to product details after new product is saved

commit 65e14ee0aebe163518ab1b243cde532019ad8af8
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 13:36:19 2023 +0530

    Make `TaskViewModel` conform to `Identifiable` and stop using `offset` for identification.

commit 5787ab7736b5036b2b42aba78707031cbc41cc12
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 13:29:35 2023 +0530

    Stop using system color.

commit 1e04f018a95853a030f9fe247467dba5638d3a69
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 13:27:53 2023 +0530

    Replace magic number used for frame height.

commit 23611c345bc1d29da84dfc881137ca9f6316037b
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 13:27:04 2023 +0530

    Move private views to helper methods.

commit 00ea28f2c5b58b022f2efb8b52edca067b7d7425
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 13:21:51 2023 +0530

    Add background color for store onboarding task list.

commit f40128fb1f0ffc9ad7f39d07a6ca236d2d8be8a1
Merge: 5ed76653f0 6b580f7ea8
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 13:01:55 2023 +0530

    Merge branch 'trunk' into feat/8907-polish-store-setup-list

commit 6b580f7ea8ec95a9278517091c965bffbd6e1c62
Merge: d0d532784c d64faf3137
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Tue Feb 28 12:42:44 2023 +0700

    Merge pull request #8993 from woocommerce/feat/8918-email-login-ui

    REST API: UI for the custom WPCom email login screen of the Jetpack Setup flow

commit d0d532784c6ef92dfcea8b907390e75123d5abab
Merge: 30e07964e8 46467939fd
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Tue Feb 28 12:42:10 2023 +0700

    Merge pull request #8990 from woocommerce/feat/8913-user-permission-modal

    REST API: Check user role and show error modal if needed

commit 46467939fdbeddb7262e5dc94d946ce84ce54dac
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Tue Feb 28 11:35:37 2023 +0700

    Rename reloadRoles to checkIfUserGotSufficientRole

commit 54727fc4cee7411d31070e799dcbc00e49d98b83
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Tue Feb 28 11:32:51 2023 +0700

    Keep weak self in retrieve user callback closure

commit 5ed76653f055e97daedf53839a221bbe8096c546
Merge: e0cbf51046 30e07964e8
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Tue Feb 28 09:55:43 2023 +0530

    Merge branch 'trunk' into feat/8907-polish-store-setup-list

commit d64faf3137b7ced8ab582fe69f48e702f2c2473b
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Tue Feb 28 11:04:10 2023 +0700

    Use example.com instead of test.com for dummy domains

commit 7354cbd787e6fb6b6bb1801950180eb0514fcd52
Merge: f9a8e8acec 30e07964e8
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 19:19:12 2023 +0000

    Merge branch 'trunk' into issue/8992-fix-ipp-survey-logic

commit f9a8e8acec6c33ca327810591f0067743d674f52
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 19:18:31 2023 +0000

    8992 add IPP feedback banner logic fix to 12.6 release

commit af6ef1458ea2bfed72fc9bc95f7d39100d2a8d11
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 19:16:04 2023 +0000

    8992 Prevent firstIPP banner from showing for old orders

commit 77aa09e15ec704d2381d14f6b74a339bf99aaaae
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 18:24:15 2023 +0000

    8992 Improve display of feedback banners

    Feedback banners were not reliably shown. This… isn’t the prettiest, because we don’t have a WCPayIPP related action to use as the prompt to update the banner type, so it uses the order action’s completion handler as a bit of a side-effect.

    It’s not too bad, because the order action is the one that fetches the relevant data, and it’s only temporary anyway.

commit b6e1773f328cad13c49f19fe6988e65e93758a46
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 18:21:56 2023 +0000

    8992 Push the feedback banner to the viewcontroller

    Previously we pushed the instruction to show a feedback banner to the view, and then the view _pulled_ the banner type it should show. This was a little confusing and while it worked, it’s error prone.

    This change pushes the banner tyep along with the instruction to show the banner.

commit 1aea6213fdec04ac7029d5822a97472268518f50
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 17:11:25 2023 +0000

    Removed outdated warning about VM recreation

    I’ve confirmed that the viewModel is now recreated when we switch sites.

commit 670c0639167cfa4c73a8e7c598eff66002f45350
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 16:20:40 2023 +0000

    8992 remove unnecessary survey completionHandler

    There’s no async work in the determination of the survey source, so the completion handler was adding unnecessary boilerplate and complexity.

commit ff96204007bdf90766a53d0037709a194708d330
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 14:45:16 2023 +0000

    8992 Require receipt_url for all orders wrt IPP surveys

    All the IPP feedback survey logic should be based on WCPay IPP transactions. Previously, the results controllers did not include the required filter for the `receipt_url` metadata, so we (mostly) considered web orders as if they were IPP orders.

    The exception to that was a post-fetch filter on the recentTransactions results.

    This commit moves the filter to the predicate, and uses the predicate for both the results controllers (all WCPay IPP transactions, and recent WCPay IPP transactions.)

    We never consider web orders, and now we can’t miss the filter out of our logic by mistake.

commit 5319accd4f4b900eb3753f76af79a45480a8a5d7
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 14:41:11 2023 +0000

    8992 Update tests to include datePaid for recency

    The conditions for surveys 2 and 3 rely on `datePaid`, which we were not setting in the tests, which means that the power users test

    test_feedbackBannerSurveySource_when_there_are_more_than_ten_wcpay_orders_then_assigns_inPersonPaymentsPowerUsers_survey

    was not arranging the data correctly, as it had no recent transactions.

commit d4c3eb7d3f7f401578a62cc1e5b289f1dfd8af24
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 14:17:24 2023 +0000

    8992 Fix arrangement of receipt_url in tests

    Prior to this, the `customFields` metadata weren’t saved to the test DB, so the tests which relied on `receipt_url` weren’t setting it correctly and would always fail regardless of what we do in the implementation.

commit 5a3a804627c2b058d21453616fa7679e0e39a19b
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 14:20:48 2023 +0000

    8992 Remove duplictated `hasTransactions` conditoin

    The base `if` for the `feedbackBannerSource` now excludes any stores which have no WCPay transactions from any later clauses, so we don’t need to check this in case 2.

commit a38379d3b1656700c9a0803432de024264fdbfd5
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 13:44:25 2023 +0000

    8992 Only require COD for the first survey

    The inPersonPaymentsCashOnDelivery survey is intended to survey merchants who sell with COD (i.e. may be interested in IPP) but with a non-WCPay solution. This is supposed to target people who may switch to IPP with WCPay.

    The other two surveys should not be limited to COD merchants, because not all WCPay merchants will want to offer that as a payment option on their website, even if they do collect IPP in some circumstances.

    This also adds tests for the negative of this scenario, and when the store is in an unsupported country.

commit 5c3659ab32bbb15ce4baa24a22a3d827ed81d2a2
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 20:32:36 2023 +0700

    Always show WPCom email login after getting sufficient permission

commit 1fd5bda6301f624fea9367cddc200f786560d08c
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 13:27:41 2023 +0000

    8992 Arrange tests of firstTransaction/powerUser surveys

    Previously we did not inject the CardPresentPaymentsConfiguration, so it defaulted to an unsupported country and these tests recieved no banner in the response.

    Also, we did not pass in the storageManager, so none of the orders we created were available to the code under test.

    These tests still fail at this commit, because the logic under test (incorrectly) requires them to have COD enabled.

commit 479aa680217bffba7cdb4bcd4e626d2f877a968c
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 20:27:36 2023 +0700

    Add tests for title and subtitle of the screen

commit 04f7e67dfd95852912f75c0ea1fbea205fdecb9d
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 20:21:04 2023 +0700

    Send requiresConnectionOnly to email login view to display the correct title and subtitle

commit 4487acf8b869a73857646b6d9252f64adf629285
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 19:48:48 2023 +0700

    Add tests for WPComEmailLoginViewModel

commit 3a468720063d45ef20d46a12d464ecce6b688e21
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 12:14:25 2023 +0000

    8992 Fix Cash on Delivery banner tests

    Previously, the main assertions of the tests were not called, because the setup of the test did not include the COD payment gateway, so there was no call to the completion handler.

    This fixes the COD tests so they correctly set up the view model to return a value, by setting a COD payment gateway and injecting the CardPresentPaymentsConfiguration for an IPP supported country.

    Tests were renamed for more specificity about the conditions under test.

    Currently, test_feedbackBannerSurveySource_Cash_on_Delivery_store_in_an_IPP_country_with_only_WCPay_web_orders_is_shown_inPersonPaymentsCashOnDelivery_survey is failing because we have not yet fixed the logic to use only IPP transactions, rather than all WCPay transactions

commit d80ed37883232ddfb93c7fd557b896952ed9596b
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 12:33:47 2023 +0000

    8992 Always call onCompletion for feedbackBanner

    By allowing the survey to fall through, we missed issues in the tests. This change ensures we always call completion, to help with fixing the survey logic.

    This change causes the four “displayed banner” tests to fail:

    - test_feedbackBannerSurveySource_when_there_are_no_wcpay_orders_then_assigns_inPersonPaymentsCashOnDelivery_survey
    - test_feedbackBannerSurveySource_when_there_is_one_wcpay_order_then_assigns_inPersonPaymentsCashOnDelivery_survey
    - test_feedbackBannerSurveySource_when_there_are_less_than_ten_wcpay_orders_then_assigns_inPersonPaymentsFirstTransaction_survey
    - test_feedbackBannerSurveySource_when_there_more_than_ten_wcpay_orders_then_assigns_inPersonPaymentsPowerUsers_survey

commit 654ae264001fc85684b897b46a855d4d3f4574a4
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 19:35:34 2023 +0700

    Handle local validation for the email field

commit 3fbf3d978314aab1824dd6c245e7a6fe119698c3
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 19:11:17 2023 +0700

    Add attributed string for WPComEmailLoginView

commit 30e07964e8258f7e0dcf7b7e313d5f7588b72f6a
Merge: 510df40473 d3a68dc39c
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 12:09:50 2023 +0000

    Merge pull request #8983 from woocommerce/task/add-ttp-settings-row-for-m2

    [Mobile Payments] Add feature flag and Tap to Pay on iPhone row for TTP M2

commit d03c619e18245d6df57b350622d98cb5178144c6
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 18:28:48 2023 +0700

    Add email field to WPComEmailLoginView

commit f2d17e43280b19b8adfc5433c32db8364e1885dd
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 11:23:26 2023 +0000

    Remove noisy teardown code

    Setting the `selectedSiteSettings` takes time, and we already replace them in `setup` anyway.
    Other properties are also replaced in `setup`
    Resetting the `storageManager` adds console noise (as the CD stack is destroyed) and the storageManager is replaced anyway.

commit d3a68dc39ce914e59327faaa647e89d2ffa0e8dc
Merge: 18a53c3e14 510df40473
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Mon Feb 27 11:08:46 2023 +0000

    Merge branch 'trunk' into task/add-ttp-settings-row-for-m2

commit e0cbf51046766c3f2ebcf9b75e8bf0380ff90581
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Mon Feb 27 16:18:44 2023 +0530

    Use system gray for background color.

commit 510df4047353b5a6812218eb21980ab96986280b
Merge: 282e17a017 8d8ffe1a71
Author: César Vargas Casaseca <cesar.vargas.casaseca@automattic.com>
Date:   Mon Feb 27 10:49:54 2023 +0100

    Merge pull request #8946 from woocommerce/task/8939-iap-order-request-original-transaction-id

    [In-App Purchases] Adapt to the new MobilePay create order request with original_transaction_id

commit 4f30f3acf7c5b147be37cc00af6390786eeaff7c
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Mon Feb 27 14:02:17 2023 +0530

    Store setup card - Add dismiss button and vertical padding with gray background.

commit 2f87620321d0813f43d1028e67edbb872dbb9d97
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 14:35:13 2023 +0700

    Present WPComEmailLogin from JetpackSetupCoordinator

commit bea6d919d254e4f1ce343da1dcceb7283cc69317
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 14:34:59 2023 +0700

    Add WPComEmailLoginHostingController

commit ebeb0cdc912877cee5c082a0ea4492ada1c35ce4
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 14:34:51 2023 +0700

    Add onSubmit closure for WPComEmailLoginViewModel

commit a3a5e05b4700f07c45991843d798f8e3fa87f5bc
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 13:50:31 2023 +0700

    Add new view and view model for WPCom email login screen

commit 2c34b8e55e0313018d62c221ff78027bf28dc34e
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Mon Feb 27 12:12:22 2023 +0530

    Revert "ReadSize extension for getting total available size in SwiftUI views."

    This reverts commit 19e0087766ecbdc8946191571b660a805870790b.

commit c39e49e84c6409e9405cb2066ac60d55b75b19c8
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Mon Feb 27 12:12:10 2023 +0530

    Revert "Use percentage based width for progress view."

    This reverts commit cb306678a9cbb868bbc0adfda39fb335dd0fbef0.

commit 3b3eaddfb27ef2f925ec63b37b876af846b34479
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Mon Feb 27 11:14:10 2023 +0530

    Show only 3 store setup tasks in collapsed mode.

commit 2e1c9f2bad17cfd44344a70570af42754728db17
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 12:26:43 2023 +0700

    Add tests for reloadRoles

commit 4ff7f4f605cd10b5a636154e22ac1c53e081091a
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 12:26:25 2023 +0700

    Make User copiable and generate code

commit 810c070cbe272817f320ce816a29023e34e34e78
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 12:02:35 2023 +0700

    Add test for roleName

commit 26631b1025541994b4a41890462155838a86412f
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 12:00:28 2023 +0700

    Add unit test for AdminRoleRequiredViewModel

commit 673419713f1e2d74d9ab2d0fa3240f42c690eec0
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 11:41:58 2023 +0700

    Switch to ScrollableVStack

commit 47ca070d240ba8b4018b18361c1664dc54bd47a5
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 11:26:54 2023 +0700

    Make AdminRoleRequiredView scrollable vertically if needed

commit 4ad9f42c83653804460a40b0dab52309fa9b4292
Merge: cb306678a9 282e17a017
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Mon Feb 27 09:55:43 2023 +0530

    Merge branch 'trunk' into feat/8907-polish-store-setup-list

commit 282e17a017a79cb5f6bad961593eda3776b79634
Merge: 356f1c2997 d583569661
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Mon Feb 27 10:17:43 2023 +0700

    Merge pull request #8972 from woocommerce/feat/8912-benefit-modal-wording

    REST API: Update Jetpack benefit modal for non-Jetpack sites

commit 356f1c299724df1ca852d83f77f7397c245958b7
Merge: b43b3762eb 0db750ac0b
Author: Rachel McR <rachel@automattic.com>
Date:   Sat Feb 25 10:35:09 2023 +0000

    Merge pull request #8965 from woocommerce/issue/8837-spe-new-product-as-draft

    SPE: Save new product as draft

commit b43b3762ebe3f4f63d42d5e34b2665931c8dd25f
Merge: 287180170d e209235f43
Author: Rachel McR <rachel@automattic.com>
Date:   Sat Feb 25 08:56:09 2023 +0000

    Merge pull request #8964 from woocommerce/issue/8837-spe-modal-product-screen

commit 287180170d723791f01a21a2ee6e885505b3bc6f
Merge: 9f19b09ea6 f6da3ac6c1
Author: Spencer Transier <17955542+spencertransier@users.noreply.github.com>
Date:   Fri Feb 24 15:40:15 2023 -0700

    Merge pull request #8986 from woocommerce/merge/12.5-code-freeze-into-trunk

    Merge 12.5 code freeze into trunk

commit f6da3ac6c1c17b281e30f830015fe1adb0eaa54a
Author: Spencer Transier <spencer@transier.us>
Date:   Fri Feb 24 14:20:05 2023 -0700

    Freeze strings for localization

commit 61e155d4033e27644affbfee7e543ec9222a43f8
Author: Spencer Transier <spencer@transier.us>
Date:   Fri Feb 24 14:19:35 2023 -0700

    Update Release Toolkit to `7.0.0`

commit e1ec7d649be21a603147c8acd29f920c1999b717
Author: Spencer Transier <spencer@transier.us>
Date:   Fri Feb 24 14:17:05 2023 -0700

    Updated Fastlane to `2.212.1`

commit ca88b163f64e0680059cb9fde214130f7bb0646f
Author: Spencer Transier <spencer@transier.us>
Date:   Fri Feb 24 14:16:34 2023 -0700

    Update metadata strings

commit 7bfb21944bd33fabc4efee78535f0a4f536f462e
Author: Spencer Transier <spencer@transier.us>
Date:   Fri Feb 24 14:16:16 2023 -0700

    Update release notes for 12.5

commit 53dce8f01c1032183f4fccf9bf2ddbf246e5ccb5
Author: Spencer Transier <spencer@transier.us>
Date:   Fri Feb 24 14:15:02 2023 -0700

    Release Notes: add new section for next version (12.6)

commit 04e00222ef367b8e489ab982cd3742384baa5f38
Author: Spencer Transier <spencer@transier.us>
Date:   Fri Feb 24 14:15:02 2023 -0700

    Update draft release notes for 12.5.

commit 26b7e6c1fa76b510e26226ea82d6aa2ed705af88
Author: Spencer Transier <spencer@transier.us>
Date:   Fri Feb 24 14:15:01 2023 -0700

    Bump version number

commit 9f19b09ea650f50e876259dce78448b8d576a623
Merge: 757d175f26 3d91a5b6dd
Author: Petros Paraskevopoulos <petros.g.paraskevopoulos@gmail.com>
Date:   Fri Feb 24 23:14:01 2023 +0200

    Merge pull request #8979 from woocommerce/changelog-12-5

    Added release notes for 12.5

commit 757d175f266cc3f6c52dbb82fc413df76a687bd2
Merge: 0cd749134e b973806720
Author: Spencer Transier <17955542+spencertransier@users.noreply.github.com>
Date:   Fri Feb 24 14:06:34 2023 -0700

    Merge pull request #8981 from woocommerce/merge/12.4-final-into-trunk

    Merge 12.4 final into trunk

commit 18a53c3e148b46e115d102fae33ef4a631af08b0
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Fri Feb 24 17:24:38 2023 +0000

    Added payments menu cell reuse fix to 12.6 release notes

commit 33389e7a6741e11d3faf8cd217d07315ed109e86
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Fri Feb 24 17:13:34 2023 +0000

    Fix incorrect disabled styling issue on cell reuse

commit 07a74a0ff1dc780b548fda7da36913164cfc9756
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Fri Feb 24 17:05:44 2023 +0000

    Hide TTP row in countries where it’s not supported

commit b8476192a43d039b5ccd5717a77977cd61d5e58d
Author: rachelmcr <rachel@automattic.com>
Date:   Fri Feb 24 14:41:21 2023 +0000

    Set product downloadable property based on whether it has downloadable files

commit 8fe1bb0fb1e94677b426cbf3976043cad7b425de
Author: rachelmcr <rachel@automattic.com>
Date:   Fri Feb 24 14:40:42 2023 +0000

    Make Downloadable Files always available when feature flag is enabled

commit e416f691785930e31e1ed1eb057e3f31e15409eb
Author: rachelmcr <rachel@automattic.com>
Date:   Fri Feb 24 13:06:17 2023 +0000

    Hide downloadable product setting when feature flag is enabled

commit e6edad0b8de1c09635a105947172bfe444ad5676
Author: rachelmcr <rachel@automattic.com>
Date:   Fri Feb 24 12:52:27 2023 +0000

    Remove Downloadable product type label for simple products

commit 1ad35cf8b3aa49fcd8e22726c2e49c97be80052a
Author: Josh Heald <joshheald@users.noreply.github.com>
Date:   Fri Feb 24 14:40:38 2023 +0000

    Add feature flag and TTPoI row for TTP M2

commit cb306678a9cbb868bbc0adfda39fb335dd0fbef0
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Fri Feb 24 19:55:47 2023 +0530

    Use percentage based width for progress view.

commit 19e0087766ecbdc8946191571b660a805870790b
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Fri Feb 24 19:55:00 2023 +0530

    ReadSize extension for getting total available size in SwiftUI views.

commit d61bfc47e03f6c8d5cc7cf3f76bc29b1577c070c
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Fri Feb 24 20:29:54 2023 +0700

    Handle reloading roles for AdminRoleRequiredViewModel

commit 0cd749134ebc5c580d1874d39b4eb16979f3b790
Merge: 594c16b79d 27adc006d6
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Fri Feb 24 21:08:14 2023 +0800

    Merge pull request #8942 from woocommerce/td/1911-replace-XLPagerTabStrip

    Dashboard stats: replace XLPagerTabStrip with WPiOS implementation

commit 594c16b79d61398f836b1eaac000628d26526c96
Merge: 2977b0667d b7e5c76f7a
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Fri Feb 24 18:03:39 2023 +0530

    Merge pull request #8948 from woocommerce/feat/8907-polish-store-setup-card

    [Store onboarding] Polish store setup card

commit b973806720150bb6c75070b4135e52a7291ea5be
Author: Spencer Transier <spencer@transier.us>
Date:   Fri Feb 24 05:03:40 2023 -0700

    Bump version number

commit e805b9b8ea07605d4681fc2b839448966af37643
Author: Spencer Transier <spencer@transier.us>
Date:   Fri Feb 24 05:03:29 2023 -0700

    Update app translations – `Localizable.strings`

commit d1f603f7f5ad00b868438a73620fc69984701d9a
Author: Spencer Transier <spencer@transier.us>
Date:   Fri Feb 24 05:02:06 2023 -0700

    Update metadata translations

commit 91b62be701c9262ce1858904bc0a59e93a8e5d1c
Author: Spencer Transier <spencer@transier.us>
Date:   Fri Feb 24 05:02:02 2023 -0700

    Update app translations – `Localizable.strings`

commit b7e5c76f7ae7c9dc49cf8686e60fa2f53c987aaa
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Fri Feb 24 17:30:48 2023 +0530

    Revert "Use `GeometryReader` and percentage based width for ProgressView."

    This reverts commit eb7a2221357407f162a304587eba2e7a4c940322.

commit 25e9daab01d56eb77e2762ee26d8ffebb34b94f4
Author: rachelmcr <rachel@automattic.com>
Date:   Fri Feb 24 11:50:45 2023 +0000

    Update release notes

commit a79bbb75d0868a954ef9280d277254cbf8b0f24d
Author: rachelmcr <rachel@automattic.com>
Date:   Fri Feb 24 11:26:35 2023 +0000

    Show disclosure indicators in product detail rows when row is actionable

commit 78f0a680e96051415f63dae965b0e0d749ca8276
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Fri Feb 24 17:56:40 2023 +0700

    Display AdminRoleRequiredView from JetpackSetupCoordinator

commit 25e67d491b5e6a4876cd043ccef32c760bec0d1f
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Fri Feb 24 17:56:28 2023 +0700

    Add new view for AdminRoleRequiredView

commit c854dcea5c5c56d47d45de139a48385644a870e7
Merge: cfa31e2575 3b16362099
Author: Rachel McR <rachel@automattic.com>
Date:   Fri Feb 24 10:47:17 2023 +0000

    Merge pull request #8977 from woocommerce/fix/revert-support-requests-release

    Beta Fix: Disable Support Request Feature

commit 3d91a5b6dd0fbdbb278cdf267339917804396f15
Author: Paolo Musolino <info@codeido.com>
Date:   Fri Feb 24 11:41:33 2023 +0100

    Update CHANGELOG.md

commit a4b2c89542d216bb6f0cd0e9de325d98b9cb66af
Author: rachelmcr <rachel@automattic.com>
Date:   Fri Feb 24 10:34:30 2023 +0000

    Update placeholder when no product category selected

commit 3b16362099891f5076be0401f51661c788921425
Author: rachelmcr <rachel@automattic.com>
Date:   Fri Feb 24 10:06:32 2023 +0000

    Revert "Enable Support Request feature flag"

    This reverts commit fc3f16d085de8e389bef9ef06f5a26fc511cb1e5.

commit f2452a45855aa27215662aeb71934f625f2d2756
Author: rachelmcr <rachel@automattic.com>
Date:   Fri Feb 24 10:06:24 2023 +0000

    Revert "Update release notes"

    This reverts commit 29939f5f0770f313d3617e21109439028f09d284.

commit 27adc006d64aaef1db315022feb695c18bb9c640
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Fri Feb 24 16:26:02 2023 +0800

    Add links to the WPiOS files in the documentation for `FilterTabBar` and `TabbedViewController`.

commit 6f9c36d4770452cd8142c9005bf3a24a8a8766fd
Merge: a4f7994e1a 79393da3dd
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Fri Feb 24 16:14:41 2023 +0800

    Merge branch 'trunk' into td/1911-replace-XLPagerTabStrip

    * trunk: (45 commits)
      Add unit tests for JetpackBenefitsViewModel
      Rename checkJetpackStatus
      Add view model for JetpackBenefitsView
      Disable profiler questions feature flag to reduce friction in the store creation flow.
      Clarify comments for AuthenticatedState
      Restore JetpackConnectionStore for authenticated state with wpcom
      Add test for JCP installation flow
      Add unit test for JetpackSetupCoordinator
      Trigger JetpackConnectionAction on the mainThread
      Fetch jetpack error from JP benefit modal
      Integrate JetpackSetupCoordinator in DashboardViewController
      Add new coordinator for Jetpack setup
      Remove redundant JetpackConnectionStore in AuthenticatedState
      Add JetpackConnectionStore to AuthenticatedState when authenticated with wporg
      Add convenience init for JetpackConnectionStore
      Support RESTRequest in RequestConverter
      Fix line length violation
      Do not store import placeholder products
      Clean child products from grouped product template
      Present single product type bottom sheet when feature flag is enabled
      ...

    # Conflicts:
    #	WooCommerce/WooCommerce.xcodeproj/project.pbxproj

commit 53a3b14e645dfcb2e16c49a8a8555f9fee4c536e
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Fri Feb 24 13:29:06 2023 +0530

    Add a completed task to the preview.

commit 8404088d17778e9c217ea5babd91ba4ba4c1ce0f
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Fri Feb 24 13:08:19 2023 +0530

    Add a note to differentiate between Tasks complete text in expanded and collapsed mode.

commit eb7a2221357407f162a304587eba2e7a4c940322
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Fri Feb 24 13:05:36 2023 +0530

    Use `GeometryReader` and percentage based width for ProgressView.

commit 1732c3bdba8fa9996bb8990b3ae38e22ca91b9c0
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Fri Feb 24 15:21:47 2023 +0800

    Add a test case on custom domains onboarding task action.

commit a7be6857767dadb00aa75718f609f3e2b210de57
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Fri Feb 24 15:11:42 2023 +0800

    Mark `start` function as `@MainActor` to avoid using a `Task` to call `start(task:)`.

commit 7818461f92eb4466cba22aaa7620c388caa7773e
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Fri Feb 24 09:31:19 2023 +0800

    Navigate to custom domains after tapping on the onboarding task.

commit d58356966157045d4277725c54d05d61a87381f7
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Fri Feb 24 14:19:22 2023 +0700

    Show benefit modal from settings for non-JP sites

commit adc77e0ac130dadaf5d473f1c703effda179d076
Author: Huong Do <hi.huongdo@gmail.com>
Date:   Fri Feb 24 14:15:26 2023 +0700

    Update wording for Jetpack benefit modal for non-JCP sites

commit dac2df6182e209dbc1aee5b2df0a29416baa7c43
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Fri Feb 24 12:12:49 2023 +0530

    Unit test `numberOfTasksCompleted` computed property of `StoreOnboardingViewModel`

commit ef5b2ec11a259f11286faf3231dd6324a34a42f3
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Fri Feb 24 12:12:41 2023 +0530

    Inject `taskViewModels` for unit testing.

commit 6ff2dca545b7c77567db9d5f3b38d13ea2ab0e0d
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Fri Feb 24 11:50:00 2023 +0530

    Use snake case for unit test name.

commit 8327b035bcee4dfdae12d3b771f6f20e0b504d02
Author: rachelmcr <rachel@automattic.com>
Date:   Thu Feb 23 23:18:57 2023 +0000

    Dismiss modals together for smoother animation

commit abe67a358c4738ce636b9d45ecfe8b14d774cbad
Author: rachelmcr <rachel@automattic.com>
Date:   Thu Feb 23 19:09:00 2023 +0000

    Dismiss modal and navigate to product details after new product is saved

commit 0db750ac0bf859f32ced022483787bb1335290dc
Author: rachelmcr <rachel@automattic.com>
Date:   Thu Feb 23 22:40:49 2023 +0000

    Fix unit test

commit 17d5c73114d4b67d0aafe8149c3a01b4664a8bf9
Author: rachelmcr <rachel@automattic.com>
Date:   Thu Feb 23 13:07:06 2023 +0000

    Create new products with draft status when feature flag is enabled

commit 59e88af1b50d914ffe32a5e6a5104e445a6d12df
Author: rachelmcr <rachel@automattic.com>
Date:   Thu Feb 23 13:06:41 2023 +0000

    Only show the Save button for new products when feature flag is enabled

commit e209235f43fd73af561ba4da0baa7ece7e4683a5
Author: rachelmcr <rachel@automattic.com>
Date:   Thu Feb 23 10:24:19 2023 +0000

    Present product form modally with feature flag enabled

commit 9e86551350046b46828c068662c0ef46c7bdf29a
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Thu Feb 23 19:01:30 2023 +0530

    Polish task list view to match UI design.

commit 57c5586f21243d85d3826ca224b8c6e990cf0062
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Thu Feb 23 18:30:40 2023 +0530

    Polish task view to match UI design.

commit 209efd16b60d3f19bb199a13216480424e7e24c7
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Thu Feb 23 18:29:19 2023 +0530

    Add computed property to calculate number of completed tasks.

commit bc2571ae3569af4f715e15d7327de53461f00241
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Thu Feb 23 18:28:33 2023 +0530

    Use the new launch store image for task list.

commit 6ec54264394fb7240fc37cfd615208b3df27e37f
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Thu Feb 23 18:21:20 2023 +0530

    Introduce a view for store setup progress view.

commit b2ce7b2c9b1947c95d7f3b4086de84b74b83af8d
Author: Sharma Elanthiraiyan <sharma.elanthiraiyan@automattic.com>
Date:   Thu Feb 23 18:12:16 2023 +0530

    Add launch store image.

commit 8d8ffe1a711163c5a069d6236e9ea10bb8d8ce39
Author: César Vargas Casaseca <cesar.vargas.casaseca@automattic.com>
Date:   Thu Feb 23 10:36:28 2023 +0100

    Adapt tests to new IAP requests approach.

commit 1c2648d85e3eec690fad6624ea7f0b538f6f9745
Author: César Vargas Casaseca <cesar.vargas.casaseca@automattic.com>
Date:   Thu Feb 23 09:41:16 2023 +0100

    Remove non necessary code

commit a4f7994e1a3025a6bc0f6dc9ea13511cc8d1c10c
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Thu Feb 23 16:22:19 2023 +0800

    Update release notes.

commit cf64ddc54199f3a7d57b64b7c502c7d6d46508fa
Author: César Vargas Casaseca <cesar.vargas.casaseca@automattic.com>
Date:   Thu Feb 23 09:17:56 2023 +0100

    Adapt request parameters to new requirements

commit 5feb403745027bbaffc710aa70be836015f8a23a
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Thu Feb 23 13:11:08 2023 +0800

    Update time range tab selection from searching for cells to buttons in UI tests.

commit ea4152f2ed5a7974f05fc3683cbe6b1dc00dad4e
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Thu Feb 23 11:17:47 2023 +0800

    Make `selection` the source of truth and only observe selected time range after the initial value is available.

commit ed794334f81c9fd9849aaf7adb8d3f2aadfb7824
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Thu Feb 23 11:05:51 2023 +0800

    Update tab bar and button styles to match the previous design as much as possible, using button configuration to fix iOS 15+ warnings.

commit c68b19c4b1c8a60c48dcd5fd3a8c1f5728f2b65e
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Wed Feb 22 22:53:54 2023 +0800

    Fix issue where the programmatically selected tab isn't showing the right child controller.

commit 08306e8a4548f206b6349d4d87f758ee3daa7fe0
Merge: 125fe81f7e 2ab993b5ee
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Wed Feb 22 14:34:44 2023 +0800

    Merge branch 'trunk' into td/1911-replace-XLPagerTabStrip

    * trunk:
      Capitalize productMultiSelectionM1  FeatureFlag
      Update Experiments/Experiments/DefaultFeatureFlagService.swift
      Update error message when add options cell can't be configured
      Update name for add options unit test
      Add product multi-selection feature flag
      Show published alert for new product flow
      Update product form notice to display status-based feedback
      Rename flag var
      Add test
      Add feature flag condition
      Add options CTA buttons to the product form
      Add feature flag condition for product form background
      Update product form background
      Make top section separators on product form full-width
      Make bottom section separators on product form full-width

commit 125fe81f7e798324fa12531741b09c8de208d3fd
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Tue Feb 21 16:37:38 2023 +0800

    Sync `selectedTimeRangeIndex` to the superclass' `selection` from tab index.

commit ea376ca54441eb2cbf8f2692de217f3e683edd04
Merge: 8065ba6157 6f573c15e0
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Tue Feb 21 15:39:17 2023 +0800

    Merge branch 'trunk' into td/1911-replace-XLPagerTabStrip

    * trunk:
      Store onboarding improvements based on CR suggestions.
      Fix test case name from CR suggestion.
      Disable `dashboardOnboarding` feature flag during UI tests.
      Add comments to `StoreOnboardingCoordinator`.
      Update font styles for the store onboarding tasks.
      Add comments to store onboarding. Remove `storeDetails` task after product confirmation.
      Update onboarding task view to a button with highlighted state with barebone navigation.
      Polish onboarding task UI.
      Show/hide onboarding UI in the dashboard based on the feature flag.
      Remove unused code from deprecated design with settings CTA on the dashboard.
      Barebone onboarding classes for UI and coordinator.
      Add the domains icon from Gridicons.
      Add a feature flag for store onboarding on the dashboard.
      Only show domain settings for admin users.

    # Conflicts:
    #	WooCommerce/WooCommerce.xcodeproj/project.pbxproj

commit 8065ba6157b4f451326d293d6c31147d07048c4b
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Tue Feb 21 13:54:05 2023 +0800

    Remove unused code from `XLPagerTabStrip`.

commit 83d143b5e00074b8ac6f7085ee0bf5b4f3341683
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Tue Feb 21 13:49:16 2023 +0800

    Update `StoreStatsAndTopPerformersViewController` to subclass `TabbedViewController` instead of `ButtonBarPagerTabStripViewController`.

commit d01d6b23bb2c677a7ec790d591ce3ae2f0b84589
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Tue Feb 21 11:43:36 2023 +0800

    Update WPiOS files to make the build pass.

commit 1d7d09b9f86338338b1647654d1f99bdee67dd02
Author: Jaclyn Chen <watertranquil@gmail.com>
Date:   Tue Feb 21 11:24:53 2023 +0800

    Add `FilterTabBar` files from WPiOS without any changes.

reginabally committed:
Add this PR to the release notes.

reginabally committed:
Merge branch 'trunk' into issue/7983-contact-customer-cta

reginabally commented:
Thanks, @rachelmcr! I committed the change with the first approach because it's more straightforward to me. Appreciate your time in reviewing the code again. 🙇🏻‍♀️

rachelmcr commented on WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressFormViewController.swift:
```diff
@@ -38,30 +38,38 @@ final class ShippingLabelAddressFormViewController: UIViewController {
-                actionSheet.addDefaultActionWithTitle(Localization.contactActionCall) { _ in
-                    if PhoneHelper.callPhoneNumber(phone: phoneNumber) == false {
-                        self.displayErrorNotice(title: Localization.phoneNumberErrorNotice)
+            if PhoneHelper.canCallPhoneNumber(phone: "123456789") == false
```
This has a (fake) string literal in it. Do you mean to check `phone` there?

```suggestion
            if PhoneHelper.canCallPhoneNumber(phone: phone) == false
```

rachelmcr commented on WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressFormViewController.swift:
```diff
@@ -38,30 +38,38 @@ final class ShippingLabelAddressFormViewController: UIViewController {
-                actionSheet.addDefaultActionWithTitle(Localization.contactActionCall) { _ in
-                    if PhoneHelper.callPhoneNumber(phone: phoneNumber) == false {
-                        self.displayErrorNotice(title: Localization.phoneNumberErrorNotice)
+            if PhoneHelper.canCallPhoneNumber(phone: "123456789") == false
+                && MFMailComposeViewController.canSendMail() == false
+                && MFMessageComposeViewController.canSendText() == false {
```
nit: This works, but just to offer another way to check that these values are `false`:

```
            if !PhoneHelper.canCallPhoneNumber(phone: "123456789")
                && !MFMailComposeViewController.canSendMail()
                && !MFMessageComposeViewController.canSendText() {
```

You could also use a `guard` statement here, rather than an `if`/`else`, if you want to avoid nesting the expected behavior in an `else` statement. That's again about style more than anything, but I find that it can help to read the code when the desired or expected behavior is less nested.

Don't feel like you have to make any changes based on nits like these! I'm just offering them as alternatives to consider.

reginabally replied:
Thank you so much for these tips! I appreciate all the suggestions and nits that help me improve my code's readability and coding style! I haven't gone through the tutorial about using `guard`, but I'll look it up!

rachelmcr reviewed (commented):
Thanks @reginabally! It looks like a fake phone number snuck in to the `PhoneHelper` check you added here; let me know if I'm missing something about that.

I also left some nit comments about style, which are entirely optional suggestions.

reginabally commented:
Thanks again for reviewing, @rachelmcr!

> It looks like a fake phone number snuck in to the PhoneHelper check you added here; let me know if I'm missing something about that.

It was initially supposed to be `phone`, but since it is optional, the `canCallPhoneNumber` condition check will fail if the customer didn't input a phone number, which defeats the purpose of checking the device's capability of opening the `tel://` URL. That is the reason why I put a fake number there.

Perhaps I should do something like this?

```
if PhoneHelper.canCallPhoneNumber(phone: {
                if let phoneNumber = phone, phoneNumber.isNotEmpty {
                    return phoneNumber
                } else {
                    return "123456789" }
            }() ) == false
```

rachelmcr commented:
> the canCallPhoneNumber condition check will fail if the customer didn't input a phone number, which defeats the purpose of checking the device's capability of opening the tel:// URL.

Don't we want it to fail in that case? We don't want to show the "Call" option if there isn't a phone number to call.

reginabally committed:
Merge branch 'trunk' into issue/7983-contact-customer-cta

reginabally committed:
Remove fake number

reginabally committed:
Merge branch 'issue/7983-contact-customer-cta' of https://github.com/woocommerce/woocommerce-ios into issue/7983-contact-customer-cta

reginabally commented:
> Don't we want it to fail in that case? We don't want to show the "Call" option if there isn't a phone number to call.

You're absolutely right! I was too focused on checking the device's capability but neglected that showing the "Call" option without a number to call is pointless.

I removed the fake number and replaced it with `phone`. Thanks again for your guidance and patience, @rachelmcr! 🙇🏻‍♀️

rachelmcr commented on WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressFormViewController.swift:
```diff
@@ -38,11 +38,18 @@ final class ShippingLabelAddressFormViewController: UIViewController {
+                    && MFMessageComposeViewController.canSendText() else {
+                self.displayErrorNotice(title: Localization.contactActionError)
+                return
+            }
```
Could you walk me through how this `guard` statement should work?

Before, the `if` statement checked if all three of these actions were **not** possible. If **none** were possible, the error was displayed.

Now, it looks like the `guard` statement checks if all three of these action **are** possible. If they aren't **all** possible, the error is displayed.

reginabally replied:
> Now, it looks like the guard statement checks if all three of these action are possible. If they aren't all possible, the error is displayed.

I got all confused with the logic! I should have used the `||` operator instead of `&&` in the guard statement so that if any one of the three conditions is true, the error will not be displayed and the rest of the code outside of the guard statement will execute. The error should only be displayed when all three conditions are false.

rachelmcr reviewed (commented):
Thanks for the changes! This is shaping up nicely.

I just have a remaining concern about the `guard` statement. It can be a little tricky to switch from using `if` to `guard` while checking multiple conditions like this, so I just want to make sure it's doing the right check now.

reginabally committed:
Merge branch 'trunk' into issue/7983-contact-customer-cta

reginabally committed:
Fix incorrect condition checks with the guard statement.

rachelmcr reviewed (approved):
Nice work! It looks good and tests well for me on a device with only some supported options. Thanks for taking this on and sticking with it! 🎉

reginabally commented:
Thank you very much for tirelessly showing me how to work on the issue, @rachelmcr!
